usl_pipeline/usl_lib/usl_lib/shared/suppliers.py

```diff
@@ -0,0 +1,49 @@
+import abc
+import os
+
+
+class ChunkFilePathGenerator(abc.ABC):
+    """Abstract layer that can be used to generate paths for chunk files."""
+
+    @abc.abstractmethod
+    def generate(self, y_chunk_index: int, x_chunk_index: int) -> str:
+        """Generates path for a chunk file for given positional indices.
+
+        Args:
+            y_chunk_index: Index along the Y-axis for a chunk position.
+            x_chunk_index: Index along the X-axis for a chunk position.
+
+        Returns:
+            A path to a chunk file.
+        """
+        pass
+
+
+class PatternBasedChunkFilePathGenerator(ChunkFilePathGenerator):
+    """An implementation of a chunk file path generator based on file name pattern."""
+
+    def __init__(self, dir_path: str, file_name_pattern: str):
+        """Constructor.
+
+        Args:
+            dir_path: A directory where every chunk file path will point.
+            file_name_pattern: A pattern for chunk file name that will be used in
+                formatter together with y-axis index of a chunk (parameter {0}) and
+                x-axis index of a chunk (parameter {1}).
+        """
+        self.dir_path = dir_path
+        self.file_name_pattern = file_name_pattern
+
+    def generate(self, y_chunk_index: int, x_chunk_index: int) -> str:
+        """Generates path for a chunk file for given positional indices.
+
+        Args:
+            y_chunk_index: Index along the Y-axis for a chunk position.
+            x_chunk_index: Index along the X-axis for a chunk position.
+
+        Returns:
+            A path to a chunk file.
+        """
+        return os.path.join(
+            self.dir_path, self.file_name_pattern.format(y_chunk_index, x_chunk_index)
+        )
```

Comment:
I'm still not sure if we need to introduce a `ChunkFilePathGenerator` abstraction. Do we actually want or need to support multiple `ChunkFilePathGenerator`s or use different patterns for naming chunks? Would this abstraction be used in by other code paths outside the chunking in here? 

If we only need one single method of naming the chunks, I think we'd be fine with

```python
def split_geotiff_into_chunks(
    elevation_file_path: str,
    chunk_size: int,
    output_dir: str,
 ) -> List[pathlib.Path]:

def chunk_name(dir_path: str, y_chunk_index: int, x_chunk_index: int) -> str:
    return os.path.join(
            dir_path, f'chunk_{y_chunk_index}_{x_chunk_index}'
    )
```
and the other chunkers can all just call `chunk_name`

Reply:
We have pretty different understanding of comfortable amount of flexibility that different parts of the logic should offer to each other. My perspective is that we might not know what needs would appear in the future for each particular sub-system. So what I'm trying to achieve is some level of independence with some way of freedom to get results in more flexible form to avoid some conflicts. Example: all our chunkers would want to produce chunk files with the same "chunk_{y}_{x}" names which would require from caller code to make different temp folders for each type of data (elevation data couldn't be processed in the same folder as say buildings data). But that's fine I can follow your suggestion now and we can reconsider it later if we need.